### PR TITLE
fix: [CDS-41587]:  deploymentTemplate infrastructure tab integration

### PR DESCRIPTION
**Pull request by PratyushGar**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 20 Issues 
* Code issue 20 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10712)

File: src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateCanvasWrapper.tsx [hunk 1]
```diff
@@ -36,7 +35,11 @@ const DeploymentTemplateCanvasWrapper = (_props: unknown, formikRef: TemplateFor
       isReadonly={isReadonly}
       gitDetails={gitDetails}
     >
-      <DeploymentTemplateCanvasWithRef ref={formikRef} />
+      <DeploymentTemplateCanvasWithRef
+        ref={formikRef}
+        templateValue={defaultTo(template.spec, {})}
+        updateTemplate={onDeploymentConfigUpdate}
```
Comment (yc147) [inline src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateCanvasWrapper.tsx:41]: Remove `templateValue` & `updateTemplate`

File: src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateCanvas.tsx [hunk 2]
```diff
@@ -7,13 +7,18 @@
 
 import React from 'react'
 import type { TemplateFormRef } from '@templates-library/components/TemplateStudio/TemplateStudio'
+import type { JsonNode } from 'services/pipeline-ng'
 import { DeploymentTemplateFormWithRef } from './DeploymentTemplateForm/DeploymentTemplateForm'
 
+export interface DeploymentTemplateFormProps {
```
Comment (yc147) [inline src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateCanvas.tsx:13]: Revert this file

File: src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentTemplateForm.tsx [hunk 3]
```diff
@@ -1,24 +1,55 @@
...
+import { DeploymentInfraWrapperWithRef } from './DeploymentInfraWrapper/DeploymentInfraWrapper'
+import type { DeploymentTemplateFormProps } from '../DeploymentTemplateCanvas'
 import css from './DeploymentTemplateForm.module.scss'
 
 export enum DeploymentTemplateFormTabs {
   Infrastructure = 'Infrastructure',
   Execution = 'Execution'
 }
 
-function DeploymentTemplateForm(_props: unknown, _formikRef: TemplateFormRef) {
```
Comment (yc147) [inline src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentTemplateForm.tsx:12]: Rename this component

File: src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraWrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+  }
+  instancesListPath?: string
+  instanceAttributes?: Array<InstanceAttributeVariable>
+}
+
+export const DeploymentInfraWrapper = (
+  props: React.PropsWithChildren<DeploymentInfraWrapperProps>,
+  formikRef: any
+): JSX.Element => {
+  const { allowableTypes, readonly, templateValue, updateTemplate } = props
```
Comment (yc147) [inline src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraWrapper.tsx:56]: Move `allowableTypes` & `readyonly` also to context

File: src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraWrapper.tsx [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+          }))
+        : [],
+      fetchInstancesScript: get(templateValue, 'infrastructure.fetchInstancesScript')
+        ? templateValue?.infrastructure.fetchInstancesScript
+        : {
+            store: {
+              type: 'Inline',
+              spec: {}
+            }
+          },
```
Comment (yc147) [inline src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraWrapper.tsx:78]: Move this logic to conext

File: src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraWrapper.tsx [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+      instanceAttributes: Array.isArray(instanceAttributes)
+        ? instanceAttributes.map(variable => ({
+            ...variable,
+            id: uuid()
+          }))
+        : [{ fieldName: 'hostName', jsonPath: '', description: '', id: uuid() }]
```
Comment (yc147) [inline src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraWrapper.tsx:84]: Move this logic to conext

File: src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraWrapper.tsx [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+    debounce(
+      (changedTemplate?: any) =>
+        changedTemplate ? updateTemplate(changedTemplate) : /* istanbul ignore next */ Promise.resolve(),
+      300
+    ),
+    [updateTemplate]
+  )
+
+  const updateTemplateValue = (infraValues: DeploymentInfra): void => {
+    const updatedInfraValues = {
```
Comment (yc147) [inline src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraWrapper.tsx:111]: use `produce` here

File: src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraWrapper.tsx [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+    debounceUpdateTemplate(updatedTemplate)
+  }
+
+  return (
+    <Container className={css.infraWidgetWrapper}>
+      <Formik
+        formName="DeploymentTemplateInfraForm"
+        onSubmit={noop}
+        initialValues={getDeploymentInfraValues()}
+        validationSchema={Yup.object().shape({
```
Comment (yc147) [inline src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraWrapper.tsx:133]: Move this schema to different file

File: src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraSpecifications/DeploymentInfraSpecifications.tsx [hunk 9]
```diff
@@ -0,0 +1,430 @@
...
+        className={css.infraSections}
+        headerClassName={css.headerText}
+      >
+        <Layout.Horizontal spacing="large">
+          <MultiTypeFieldSelector
+            name="variables"
+            label={getString('pipeline.customDeployment.infraVariablesTitle')}
+            defaultValueToReset={[]}
+            disableTypeSelection
+            style={{ flexGrow: 1, marginBottom: 0 }}
```
Comment (yc147) [inline src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraSpecifications/DeploymentInfraSpecifications.tsx:161]: avoid inline style

File: src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraSpecifications/DeploymentInfraSpecifications.tsx [hunk 10]
```diff
@@ -0,0 +1,430 @@
...
+              setFieldValue(`variables[${index}].default`, defaultValue)
+            }}
+            isReadonly={readonly}
+          />
+        ) : null}
+      </>
+    )
+  }
+
+  const FetchScriptWidgetTitle = useCallback(
```
Comment (yc147) [inline src/modules/75-cd/components/TemplateStudio/DeploymentTemplateCanvas/DeploymentTemplateForm/DeploymentInfraWrapper/DeploymentInfraSpecifications/DeploymentInfraSpecifications.tsx:137]: use `useMemo`

File: src/modules/75-cd/components/DeploymentTemplate/DeploymentTemplate.tsx [hunk 11]
```diff
@@ -19,7 +19,7 @@ export class DeploymentTemplate extends Template {
   protected type = TemplateType.CustomDeployment
   protected label = 'Deployment'
   protected color = Color.LIME_500
```
Comment (yc147) [inline src/modules/75-cd/components/DeploymentTemplate/DeploymentTemplate.tsx:21]: Remove `color`

**Review by yc147 — APPROVED** [review]

**Pull request merged**